File: tests/rbac/fixtures/identity_groups.ts
export interface IdentityGroupFixtureMember {
  readonly subject: string;
  readonly email: string;
  readonly displayName: string;
  readonly active: boolean;
}

export interface IdentityGroupFixture {
  readonly providerId: string;
  readonly externalGroupId: string;
  readonly displayName: string;
  readonly immutableId: string;
  readonly members: readonly IdentityGroupFixtureMember[];
}

export const IDENTITY_GROUP_FIXTURES: readonly IdentityGroupFixture[] = [
  {
    providerId: "oidc-claims",
    externalGroupId: "eng-platform-admins",
    immutableId: "gid-eng-platform-admins",
    displayName: "Engineering Platform Admins",
    members: [
      {
        subject: "alice-admin-sub",
        email: "alice_admin@example.test",
        displayName: "Alice Admin",
        active: true,
      },
    ],
  },
  {
    providerId: "oidc-claims",
    externalGroupId: "eng-platform-users",
    immutableId: "gid-eng-platform-users",
    displayName: "Engineering Platform Users",
    members: [
      {
        subject: "bob-chat-user-sub",
        email: "bob_chat_user@example.test",
        displayName: "Bob Chat User",
        active: true,
      },
      {
        subject: "dave-no-role-sub",
        email: "dave_no_role@example.test",
        displayName: "Dave No Role",
        active: true,
      },
    ],
  },
  {
    providerId: "okta-primary",
    externalGroupId: "kb-ingestors",
    immutableId: "00g-kb-ingestors",
    displayName: "Knowledge Base Ingestors",
    members: [
      {
        subject: "carol-kb-ingestor-sub",
        email: "carol_kb_ingestor@example.test",
        displayName: "Carol KB Ingestor",
        active: true,
      },
    ],
  },
];

export const IDENTITY_GROUP_SYNC_RULE_FIXTURE = {
  id: "rule-platform-groups",
  providerId: "oidc-claims",
  name: "Platform groups",
  priority: 10,
  includePatterns: ["^Engineering Platform (?<role>Admins|Users)$"],
  excludePatterns: ["^Engineering Platform Contractors$"],
  teamNameTemplate: "Platform",
  teamSlugTemplate: "platform",
  roleMap: {
    Admins: "admin",
    Users: "member",
  },
  autoCreateTeam: true,
} as const;

File: tests/rbac/fixtures/identity_groups.py
"""Representative identity group fixtures for RBAC tests."""

from __future__ import annotations

from typing import Final, TypedDict


class IdentityGroupFixtureMember(TypedDict):
    """Member record carried by an upstream identity group fixture."""

    subject: str
    email: str
    display_name: str
    active: bool


class IdentityGroupFixture(TypedDict):
    """Upstream group record used by identity sync tests."""

    provider_id: str
    external_group_id: str
    immutable_id: str
    display_name: str
    members: tuple[IdentityGroupFixtureMember, ...]


IDENTITY_GROUP_FIXTURES: Final[tuple[IdentityGroupFixture, ...]] = (
    {
        "provider_id": "oidc-claims",
        "external_group_id": "eng-platform-admins",
        "immutable_id": "gid-eng-platform-admins",
        "display_name": "Engineering Platform Admins",
        "members": (
            {
                "subject": "alice-admin-sub",
                "email": "alice_admin@example.test",
                "display_name": "Alice Admin",
                "active": True,
            },
        ),
    },
    {
        "provider_id": "oidc-claims",
        "external_group_id": "eng-platform-users",
        "immutable_id": "gid-eng-platform-users",
        "display_name": "Engineering Platform Users",
        "members": (
            {
                "subject": "bob-chat-user-sub",
                "email": "bob_chat_user@example.test",
                "display_name": "Bob Chat User",
                "active": True,
            },
            {
                "subject": "dave-no-role-sub",
                "email": "dave_no_role@example.test",
                "display_name": "Dave No Role",
                "active": True,
            },
        ),
    },
    {
        "provider_id": "okta-primary",
        "external_group_id": "kb-ingestors",
        "immutable_id": "00g-kb-ingestors",
        "display_name": "Knowledge Base Ingestors",
        "members": (
            {
                "subject": "carol-kb-ingestor-sub",
                "email": "carol_kb_ingestor@example.test",
                "display_name": "Carol KB Ingestor",
                "active": True,
            },
        ),
    },
)

IDENTITY_GROUP_SYNC_RULE_FIXTURE: Final = {
    "id": "rule-platform-groups",
    "provider_id": "oidc-claims",
    "name": "Platform groups",
    "priority": 10,
    "include_patterns": (r"^Engineering Platform (?P<role>Admins|Users)$",),
    "exclude_patterns": (r"^Engineering Platform Contractors$",),
    "team_name_template": "Platform",
    "team_slug_template": "platform",
    "role_map": {
        "Admins": "admin",
        "Users": "member",
    },
    "auto_create_team": True,
}

__all__ = [
    "IDENTITY_GROUP_FIXTURES",
    "IDENTITY_GROUP_SYNC_RULE_FIXTURE",
    "IdentityGroupFixture",
    "IdentityGroupFixtureMember",
]
